infogami/infobase/core.py

```diff
@@ -115,7 +115,12 @@ def get(self, key, default=None):
             return default
 
     def __repr__(self):
-        return "<thing: %s>" % repr(self.key)
+        # Use u'xyz' for strings, for consistency across Python 2 & 3
```

Comment:
This exchanges speed now for a bunch of stuff that we will need to undo in the future.  I would rather test with `pytest --doctest-modules ` and mark these instances with `# doctest: +ALLOW_UNICODE` which is easy to find and fix up when we drop legacy Python.

Reply:
The problem as I understand it is that `doctest: +ALLOW_UNICODE` makes sure that

> u'hello'

is permitted when a doctest permits 'hello'.

This line isn't about that; it's about `<thing u'/type/string'>`, which is not the same as `u'/type/string'.

Can discuss more. The simplest forward-movement-minded fix is to invert the logic -- on PY2, if it's a Unicode string, print it without the leading `u`. I'm happy to do that, but I think we need one or the other, and that `doctest: +ALLOW_UNICODE` won't save us here.

Reply:
If the test expects `u'hello' then it passes on Py3 if the result is `'hello'` because all str are Unicode in Py3.

I totally rewrote that one test in 90 like https://github.com/internetarchive/infogami/pull/90/files#diff-809b4ebcf7b0961aa16de6c2bce7e553R132-R142 ...  It is more verbose but test all the same things.